src/neuroglancer/datasource/swc/frontend.ts

```diff
@@ -0,0 +1,551 @@
+//   }
+//   get volumeType() {
+//     return this.info.volumeType;
+//   }
+
+//   get rank() {
+//     return 3;
+//   }
+
+//   constructor(
+//       chunkManager: ChunkManager, public baseUrl: string, public nodeKey: string,
+//       public dataInstanceKey: string, public info: VolumeDataInstanceInfo) {
+//     super(chunkManager);
+//   }
+
+//   getSources(volumeSourceOptions: VolumeSourceOptions) {
+//     return this.info.getSources(
+//         this.chunkManager, {
+//           'baseUrl': this.baseUrl,
+//           'nodeKey': this.nodeKey,
+//           'dataInstanceKey': this.dataInstanceKey,
+//         },
+//         volumeSourceOptions);
+//   }
+// }
+
+const urlPattern = /^((?:http|https):\/\/[^\/]+)\/([^\/]+)\/([^\/]+)$/;
+
+export function getDataSource(options: GetDataSourceOptions): Promise<DataSource> {
+  let match = options.providerUrl.match(urlPattern);
+  if (match === null) {
+    throw new Error(`Invalid SWC URL: ${JSON.stringify(options.providerUrl)}.`);
+  }
+  const baseUrl = match[1];
+  const nodeKey = match[2];
+  const dataInstanceKey = match[3];
+  return options.chunkManager.memoize.getUncounted(
+      {
+        type: 'svc:MultiscaleVolumeChunkSource',
+        baseUrl,
+        nodeKey: nodeKey,
+        dataInstanceKey,
+      },
+      async () => {
+        // const serverInfo = await getServerInfo(options.chunkManager, baseUrl);
+        // let repositoryInfo = serverInfo.getNode(nodeKey);
+        // if (repositoryInfo === undefined) {
+        //   throw new Error(`Invalid node: ${JSON.stringify(nodeKey)}.`);
+        // }
+        // const dataInstanceInfo = repositoryInfo.dataInstances.get(dataInstanceKey);
+        // if (!(dataInstanceInfo instanceof VolumeDataInstanceInfo)) {
+        //   throw new Error(`Invalid data instance ${dataInstanceKey}.`);
+        // }
+        // const info =
+        //     await getDataInstanceDetails(options.chunkManager, baseUrl, nodeKey, dataInstanceInfo);
+        // const volume = new SWCMultiscaleVolumeChunkSource(
+        //     options.chunkManager, baseUrl, nodeKey, dataInstanceKey, info);
+        // const box: BoundingBox = {
+        //   lowerBounds: new Float64Array(info.lowerVoxelBound),
+        //   upperBounds: Float64Array.from(info.upperVoxelBoundInclusive, x => x + 1)
+        // };
+        const modelSpace = makeCoordinateSpace({
+          rank: 3,
+          names: ['x', 'y', 'z'],
+          units: ['m', 'm', 'm'],
+          // scales: Float64Array.from(info.voxelSize, x => x / 1e9), ???
+          scales: Float64Array.from(vec3.create(), x => x / 1e9),
+          // boundingBoxes: [makeIdentityTransformedBoundingBox(box)], ???
+          boundingBoxes: [],
+        });
+        const dataSource: DataSource = {
+          modelTransform: makeIdentityTransform(modelSpace),
+          subsources: [{
+            id: 'default',
+            // subsource: {volume},
+            subsource: {},
+            default: true,
```

Comment:
Delete this empty subsource.  Instead, the skeletons one added below should be given an id of "default".